Engine/DataFeeds/DataManager.cs

```diff
@@ -170,6 +168,10 @@ public void RemoveAllSubscriptions()
         /// <returns>True if the subscription was created and added successfully, false otherwise</returns>
         public bool AddSubscription(SubscriptionRequest request)
         {
+            // guarantee the configuration is present in our config collection
+            // this is related to GH issue 3877: where we added a configuration which we also removed
+            _subscriptionManagerSubscriptions.TryAdd(request.Configuration, request.Configuration);
```

Comment:
Interesting. This change looks good and make complete sense, interesting case there removing/re-adding subscriptions. I think previously we've thrown around ideas of reference counting the subscriptions.